snowdialect.py

```diff
@@ -296,6 +301,41 @@ def get_pk_constraint(self, connection, table_name, schema=None, **kw):
         ).get(table_name, {'constrained_columns': [],
                            'name': None})
 
+    @reflection.cache
+    def _get_schema_unique_constraints(self, connection, schema, **kw):
+        result = connection.execute(text(
+            "SHOW /* sqlalchemy:_get_schema_unique_constraints */ UNIQUE KEYS IN SCHEMA {0}".format(schema)
+        ))
+        unique_constraints = {}
+        for row in result:
+            name = self.normalize_name(row['constraint_name'])
+            if name not in unique_constraints:
+                unique_constraints[name] = {
+                    'column_names': [self.normalize_name(row['column_name'])],
+                    'name': name,
+                    'table_name': self.normalize_name(row['table_name'])
+                }
+            else:
+                unique_constraints[name]['column_names'].append(self.normalize_name(row['column_name']))
+
+        ans = {}
+        for _, v in iteritems(unique_constraints):
+            if v['table_name'] not in ans:
+                ans[v['table_name']] = []
+            ans[v['table_name']].append({k2: v2 for k2, v2 in iteritems(v) if k2 != 'table_name'})
```

Comment:
Just as an FYI I think this would be a little more concise
```suggestion
        for _, v in iteritems(unique_constraints):
            ans.setdefault(v['table_name'], list()).append({k2: v2 for k2, v2 in iteritems(v) if k2 != 'table_name'})
```

Reply:
This is my minimal version, but I could have added some bugs. Please see if you liked this, I think this is easier to understand (but I know that's relative).
```suggestion
        for _, v in iteritems(unique_constraints):
            table_name = v.pop('table_name')
            ans.setdefault(v[table_name], list()).append(v)
```

Reply:
To be honest, I copied these from `_get_schema_foreign_keys`, but if we're cleaning things up, I updated to what I would consider the cleanest version :slightly_smiling_face: